metrics/web/docs/features/deep_links/diagrams/custom_approach_component_diagram.puml

```diff
@@ -0,0 +1,13 @@
+@startuml custom_approach_component_diagram
+
+skinparam componentStyle rectangle
+
+component Browser #A9DCDF
+component NavigationComponent
+component ApplicationState
```

Comment:
Would it be great to unite the Metrics Web application components into a group/package/top-level component to make this diagram a bit more clear? Also, would it be great to display a `User Interface` as a separate component here? I think it would allow us to show the full deep links data flow from the `Browser` to the `UI` and vice versa and provide a bit more clear top-level understanding of how would the deep links work.